x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java

```diff
@@ -91,15 +99,25 @@ protected void nodeOperation(AllocatedPersistentTask task, ShardFollowTask param
         ShardFollowNodeTask shardFollowNodeTask = (ShardFollowNodeTask) task;
         Client leaderClient = params.getLeaderClusterAlias() != null ?
                 this.client.getRemoteClusterClient(params.getLeaderClusterAlias()) : this.client;
-        logger.info("Starting shard following [{}]", params);
-        fetchGlobalCheckpoint(client, params.getFollowShardId(),
-                followGlobalCheckPoint -> {
-                        shardFollowNodeTask.updateProcessedGlobalCheckpoint(followGlobalCheckPoint);
-                        prepare(leaderClient, shardFollowNodeTask, params, followGlobalCheckPoint);
-                    }, task::markAsFailed);
+        IndexMetadataVersionChecker checker = new IndexMetadataVersionChecker(params.getLeaderShardId().getIndex(),
```

Comment:
I think `checker` might be better as `imdVersionChecker`?

Reply:
You mean to rename the class `IndexMetadataVersionChecker` to `Checker`?

Reply:
I mean that the variable name `checker` is unclear.